services/resources-getter.js

```diff
@@ -243,16 +251,32 @@ function ResourcesGetter(model, opts, params) {
     return getSegmentCondition()
       .then(function () {
         var query = getRecords();
+        var fieldsSearched = null;
 
         if (hasRelationshipFilter()) {
           return exec(getRecords())
             .then(function (records) {
               var count = records.length;
               records = records.slice(getSkip(), getSkip() + getLimit());
-              return [count, records];
+
+              if (params.search) {
+                fieldsSearched = searchBuilder.getFieldsSearched();
+              }
+
+              return [records, count, fieldsSearched];
             });
         } else {
-          return P.all([count(query), exec(getRecords())]);
+          return P.all([exec(getRecords()), count(query)])
+            .then(function (results) {
+              var records = results[0];
+              var count = results[1];
+
+              if (params.search) {
+                fieldsSearched = searchBuilder.getFieldsSearched();
+              }
+
+              return [records, count, fieldsSearched];
+            });
```

Comment:
This is duplicate code `if` and `else` cases have the same final behaviour. You can factorize this.

Reply:
I'll fix this ;)